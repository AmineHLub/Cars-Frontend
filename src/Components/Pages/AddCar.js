import React from 'react';

const AddCar = () => (
  <div className="container">
    <div className="add-box">
      <h1>Add a Car</h1>
      <div className="labeled">
        <input required type="text" placeholder="Name" name="name" id="name" />
      </div>
      <div className="labeled">
        <textarea required className="input bigput" placeholder="Description" maxLength="250" minLength="1" name="description" />
      </div>
      <div className="labeled">
        <input required type="text" placeholder="Model_info" name="model_info" id="model_info" />
      </div>
      <div className="labeled">
        <input required type="number" min="1" className="input linput" placeholder="Reservation Fee" id="reservation_fee" name="reservatio_fee" />
      </div>
      <div className="labeled">
        <input type="file" className="input linput" name="image" id="image" />
      </div>
      <button type="button" className="log-button add-b">Create</button>
    </div>
  </div>
);
export default AddCar;
