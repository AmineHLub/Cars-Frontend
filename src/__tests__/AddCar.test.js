import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import AddCar from '../components/Pages/AddCar';

// const testData = {
//   id: 1,
//   name: 'Freetown',
//   population: '7,000,000',
//   moreInfo: {
//     cityImage: 'test.png',
//   },
// };

describe('Testing AddCar component', () => {
  test('renders a correct data', () => {
    const { getByTestId } = render(<Router><AddCar /></Router>);
    expect(getByTestId('image')).toBeInTheDocument();
    expect(getByTestId('name')).toBeInTheDocument();
    expect(getByTestId('description')).toBeInTheDocument();
    expect(getByTestId('model-info')).toBeInTheDocument();
    expect(getByTestId('reservation-fee')).toBeInTheDocument();
  });

  test('renders as expected', () => {
    const tree = renderer.create(<Router><AddCar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
