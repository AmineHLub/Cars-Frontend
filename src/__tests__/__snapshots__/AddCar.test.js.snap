// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCar component renders as expected 1`] = `
<div
  className="absolute h-screen bg-primaryGreen w-full"
>
  <div
    className="bg-show-car opacity-20"
  />
  <div
    className="text-white opacity"
  />
  <div
    className="absolute top-0 h-screen w-full flex items-start justify-center"
  >
    <div
      className="py-4 px-4 flex flex-col items-start justify-center"
    >
      <div
        className="text-center p-4 w-full"
      >
        <h1
          className="text-white text-lg md:text-2xl md:mb-10 font-bold"
        >
          ADD A CAR
        </h1>
        <p
          className="text-white mt-5 md:text-xl"
        >
          Please fill in the form with with correct details in order to add a new car.
        </p>
      </div>
      <div
        className="mx-auto p-2 md:mt-10 md:w-full"
      >
        <form
          className="bg-white border border-white rounded-md"
          data-testid="form"
          onSubmit={[Function]}
        >
          <div
            className="border border-gray-300 flex items-center"
          >
            <input
              className="text-gray-800 pl-3 h-12 w-full"
              data-testid="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Name"
              required={true}
              type="text"
            />
          </div>
          <div
            className="border border-gray-300 flex items-center"
          >
            <input
              className="text-gray-800 pl-3 h-12 w-full"
              data-testid="description"
              name="description"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Description"
              required={true}
              type="textarea"
            />
          </div>
          <div
            className="border border-gray-300 flex items-center"
          >
            <input
              className="text-gray-800 pl-3 h-12 w-full"
              data-testid="model-info"
              name="model_info"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Model info"
              required={true}
              type="text"
            />
          </div>
          <div
            className="border border-gray-300 flex items-center"
          >
            <input
              className="text-gray-800 pl-3 h-12 w-full"
              data-testid="reservation-fee"
              min="1"
              name="reservation_fee"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Reservation Fee"
              required={true}
              type="number"
            />
          </div>
          <div
            className="border border-gray-300 flex items-center p-3"
          >
            <input
              className="text-gray-800 w-full"
              data-testid="image"
              min="1"
              name="image"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Choose File"
              required={true}
              type="file"
            />
          </div>
          <div
            className="w-full h-auto border border-gray-300 flex items-center justify-center pt-10 pb-5"
          >
            <button
              className="bg-primaryGreen border-2 border-white p-3 rounded-full w-full mx-14 md:w-1/2 text-white font-semibold cursor-pointer"
              type="submit"
            >
              Create
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
